src/rust/bitbox02/src/lib.rs

```diff
@@ -140,8 +141,8 @@ pub fn screen_print_debug(msg: &str, duration: i32) {
         Ok(cstr) => unsafe {
             bitbox02_sys::screen_print_debug(cstr.as_ptr() as *const _, duration)
         },
-        Err(cstr) => unsafe {
-            bitbox02_sys::screen_print_debug(cstr.as_ptr() as *const _, duration)
+        Err(errmsg) => unsafe {
+            bitbox02_sys::screen_print_debug(errmsg.as_ptr() as *const _, duration)
```

Comment:
errmsg is a rust string that is not null terminated, so this does not work.

Reply:
done